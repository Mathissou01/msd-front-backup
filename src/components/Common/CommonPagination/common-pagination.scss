@use "src/styles/02-tools/tools.typography" as typo;
@use "src/styles/01-settings/settings.colors" as colors;
@use "src/styles/02-tools/tools.shadows" as shadows;
@use "src/styles/02-tools/tools.focus" as focus;

.c-CommonPagination {
  display: flex;
  align-items: center;
  justify-content: center;

  &_hasRowsPerPage {
    justify-content: space-between;
  }

  &__RowsPerPage {
    display: flex;
    flex-direction: row;
    gap: 16px;
    align-items: center;
    justify-content: center;
    padding: 0;
  }

  &__Select {
    @include shadows.shadow(1);
    @include focus.focus-ring(true, 2px);

    padding: 10px;
    color: colors.$blue-1;
    background-color: #fff;
    border: hidden;
    border-radius: 5px;
  }

  &__Label {
    color: colors.$blue-1;
  }

  &__Navigation {
    display: flex;
    flex-direction: row;
    gap: 14px;
    align-items: center;
    padding: 0;
  }

  &__Pages {
    display: flex;
    flex-direction: row;
    gap: 4px;
    align-items: flex-start;
    justify-content: center;
  }

  &__Page {
    @include focus.focus-ring(true, 2px);

    padding: 4px 2px;
    color: colors.$blue-1;
    font-weight: 350;
    line-height: 18px;

    &_active {
      color: colors.$expert-blue;
      font-weight: 700;
      font-family: DINPro-Medium, sans-serif;
    }
  }

  &__Backward,
  &__Forward {
    display: flex;
    flex-direction: row;
    gap: 2px;
    align-items: flex-start;
    padding: 0;

    &_disabled {
      opacity: 0.3;
      user-select: none;
      pointer-events: none;
    }
  }

  &__Button {
    @include focus.focus-ring;

    display: flex;
    box-sizing: content-box;
    width: 24px;
    height: 24px;
    padding: 4px 3px;

    &_reverse {
      transform: rotate(180deg);
    }

    &_arrow,
    &_arrowTo {
      background-repeat: no-repeat;
      background-position: center;
      background-size: 24px;
    }

    &_arrow {
      background-image: url("data:image/svg+xml,%3Csvg width='25' height='24' viewBox='0 0 25 24' fill='none' xmlns='http://www.w3.org/2000/svg'%3E%3Cpath d='M.5 12a12 12 0 1 0 24 0 12 12 0 0 0-24 0Zm13.71-5.21 4.5 4.5a1 1 0 0 1 0 1.41l-4.5 4.5a1 1 0 0 1-1.42-1.41l2.37-2.37a.24.24 0 0 0 .05-.27.25.25 0 0 0-.23-.15H6.5a1 1 0 0 1 0-2h8.48a.259.259 0 0 0 .24-.3.26.26 0 0 0-.06-.13L12.79 8.2a1 1 0 1 1 1.42-1.41Z' fill='%23767C97'/%3E%3C/svg%3E");
    }

    &_arrowTo {
      background-image: url("data:image/svg+xml,%3Csvg width='25' height='24' viewBox='0 0 25 24' fill='none' xmlns='http://www.w3.org/2000/svg'%3E%3Cpath d='M8.488 15.658a.855.855 0 0 1-1.238-.765V9.1a.855.855 0 0 1 1.238-.765l5.789 2.894a.857.857 0 0 1 0 1.531l-5.789 2.898Z' stroke='%23767C97' stroke-width='1.5' stroke-linecap='round' stroke-linejoin='round'/%3E%3Cpath d='M1.25 11.998a11.25 11.25 0 1 0 22.5 0 11.25 11.25 0 0 0-22.5 0v0Zm16.5-3.75v7.5' stroke='%23767C97' stroke-width='1.5' stroke-linecap='round' stroke-linejoin='round'/%3E%3C/svg%3E");
    }
  }
}
