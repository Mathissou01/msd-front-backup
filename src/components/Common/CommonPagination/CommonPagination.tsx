import React from "react";
import Image from "next/image";
import classNames from "classnames";
import { makePublicAssetPath } from "../../../lib/utilities";
import "./common-pagination.scss";

export interface ICommonPaginationProps {
  currentPage: number;
  rowCount: number;
  onChangePage: (page: number, totalRows: number) => void;
  onChangeRowsPerPage?: (
    currentRowsPerPage: number,
    currentPage: number,
  ) => void;
  noRowsPerPage?: boolean;
  rowsPerPage?: number;
  rowsPerPageOptions?: Array<number>;
  rowsPerPageText?: string;
}

export default function CommonPagination({
  currentPage,
  rowCount,
  onChangePage,
  onChangeRowsPerPage,
  noRowsPerPage,
  rowsPerPage = 10,
  rowsPerPageOptions = [10, 20, 50, 100],
  rowsPerPageText = "Entrées par page",
}: ICommonPaginationProps) {
  /* Static Data */
  const navigationArrowSrc = "/images/pictos/navigation-arrow.svg";
  const navigationArrowToSrc = "/images/pictos/navigation-arrow-to.svg";
  const alternativeText = {
    arrowToFirst: "Aller à la première page",
    arrowToLast: "Aller à la dernière page",
    arrowPrevious: "Aller à la page précédente",
    arrowNext: "Aller à la page suivante",
  };

  /* Local Data */
  const pageCount = Math.ceil(rowCount / rowsPerPage);
  const paginationClassNames = classNames("c-CommonPagination", {
    "c-CommonPagination_hasRowsPerPage": !noRowsPerPage,
  });
  const previousClassNames = classNames("c-CommonPagination__Backward", {
    "c-CommonPagination__Backward_disabled": currentPage === 1,
  });
  const nextClassNames = classNames("c-CommonPagination__Forward", {
    "c-CommonPagination__Forward_disabled": currentPage === pageCount,
  });

  return (
    <div className={paginationClassNames}>
      {!noRowsPerPage && (
        <div className="c-CommonPagination__RowsPerPage">
          <select
            className="c-CommonPagination__Select"
            name="rowsPerPage"
            id="rowsPerPage"
            defaultValue={rowsPerPage}
            onChange={(t) =>
              onChangeRowsPerPage &&
              onChangeRowsPerPage(parseInt(t.target.value), currentPage)
            }
          >
            {rowsPerPageOptions?.map((option, index) => (
              <option key={index} value={option}>
                {option}
              </option>
            ))}
          </select>
          <span className="c-CommonPagination__Label">{rowsPerPageText}</span>
        </div>
      )}
      <nav className="c-CommonPagination__Navigation">
        <div className={previousClassNames}>
          <button
            type="button"
            className="c-CommonPagination__Button"
            disabled={currentPage <= 1}
            onClick={() => onChangePage(1, rowCount)}
          >
            <Image
              className="c-CommonPagination__Button_reverse"
              src={makePublicAssetPath(navigationArrowToSrc)}
              alt={alternativeText.arrowToFirst}
              height={24}
              width={24}
            />
          </button>
          <button
            type="button"
            className="c-CommonPagination__Button"
            disabled={currentPage <= 1}
            onClick={() => onChangePage(currentPage - 1, rowCount)}
          >
            <Image
              className="c-CommonPagination__Button_reverse"
              src={makePublicAssetPath(navigationArrowSrc)}
              alt={alternativeText.arrowPrevious}
              height={24}
              width={24}
            />
          </button>
        </div>
        <div className="c-CommonPagination__Pages">
          {[...Array(pageCount)].map((e, i) => (
            <button
              key={i + 1}
              className={classNames("c-CommonPagination__Page", {
                "c-CommonPagination__Page_active": i + 1 === currentPage,
              })}
              onClick={() => onChangePage(i + 1, rowCount)}
            >
              {i + 1}
            </button>
          ))}
        </div>
        <div className={nextClassNames}>
          <button
            type="button"
            className="c-CommonPagination__Button"
            disabled={currentPage >= pageCount}
            onClick={() => onChangePage(currentPage + 1, rowCount)}
          >
            <Image
              src={makePublicAssetPath(navigationArrowSrc)}
              alt={alternativeText.arrowNext}
              height={24}
              width={24}
            />
          </button>
          <button
            type="button"
            className="c-CommonPagination__Button"
            disabled={currentPage >= pageCount}
            onClick={() => onChangePage(pageCount, rowCount)}
          >
            <Image
              src={makePublicAssetPath(navigationArrowToSrc)}
              alt={alternativeText.arrowToLast}
              height={24}
              width={24}
            />
          </button>
        </div>
      </nav>
    </div>
  );
}
