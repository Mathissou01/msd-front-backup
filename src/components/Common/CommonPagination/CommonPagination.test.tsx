import { fireEvent, render, screen } from "@testing-library/react";
import CommonPagination from "./CommonPagination";
import React from "react";

const mock = {
  currentPage: 1,
  rowCount: 21,
};

describe("CommonPagination", () => {
  it("renders", () => {
    const handleClick = jest.fn();
    const { container } = render(
      <CommonPagination
        currentPage={mock.currentPage}
        rowCount={mock.rowCount}
        onChangePage={handleClick}
        onChangeRowsPerPage={handleClick}
      />,
    );
    fireEvent.click(screen.getByText(3));
    expect(handleClick).toHaveBeenCalledTimes(1);
    expect(container).toMatchSnapshot();
  });
});
