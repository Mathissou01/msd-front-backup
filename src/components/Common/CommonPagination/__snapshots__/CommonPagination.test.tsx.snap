// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommonPagination renders 1`] = `
<div>
  <div
    class="c-CommonPagination c-CommonPagination_hasRowsPerPage"
  >
    <div
      class="c-CommonPagination__RowsPerPage"
    >
      <select
        class="c-CommonPagination__Select"
        id="rowsPerPage"
        name="rowsPerPage"
      >
        <option
          selected=""
          value="10"
        >
          10
        </option>
        <option
          value="20"
        >
          20
        </option>
        <option
          value="50"
        >
          50
        </option>
        <option
          value="100"
        >
          100
        </option>
      </select>
      <span
        class="c-CommonPagination__Label"
      >
        Entr√©es par page
      </span>
    </div>
    <nav
      class="c-CommonPagination__Navigation"
    >
      <div
        class="c-CommonPagination__Backward c-CommonPagination__Backward_disabled"
      >
        <button
          class="c-CommonPagination__Button c-CommonPagination__Button_arrowTo c-CommonPagination__Button_reverse"
          disabled=""
          type="button"
        />
        <button
          class="c-CommonPagination__Button c-CommonPagination__Button_arrow c-CommonPagination__Button_reverse"
          disabled=""
          type="button"
        />
      </div>
      <div
        class="c-CommonPagination__Pages"
      >
        <button
          class="c-CommonPagination__Page c-CommonPagination__Page_active"
        >
          1
        </button>
        <button
          class="c-CommonPagination__Page"
        >
          2
        </button>
        <button
          class="c-CommonPagination__Page"
        >
          3
        </button>
      </div>
      <div
        class="c-CommonPagination__Forward"
      >
        <button
          class="c-CommonPagination__Button c-CommonPagination__Button_arrow"
          type="button"
        />
        <button
          class="c-CommonPagination__Button c-CommonPagination__Button_arrowTo"
          type="button"
        />
      </div>
    </nav>
  </div>
</div>
`;
